guides/theming-your-components.md

```diff
@@ -1,135 +1,224 @@
+can then include these mixins in your application along with Angular Material's own mixins.
+
+## Step-by-step example
+
+To illustrate participation in Angular Material's theming system, we can look at an example of a
+custom carousel component. The carousel starts with a single file, `carousel.scss`, that contains
+structural, color, and typography styles. This file is included in the `styleUrls` of the component.
+
+```scss
+// carousel.scss
+
+.my-carousel {
+  display: flex;
+  font-family: serif;
+}
+
+.my-carousel-button {
+  border-radius: 50%;
+  color: blue;
 }
 ```
 
-Finally, create a mixin that accepts an Angular Material theme, and delegates to the individual
-theming system mixins based on the configurations. A theme consists of configurations for
-individual theming systems (`color` and `typography`).
+### Step 1: Extact theme-based styles to a separate file
+
+To change this file to participate in Angular Material's theming system, we split the styles into
+two files, with the color and typography styles moved into mixins. By convention, the new file
+name ends with `-theme`. Additionally, the file starts with an underscore (`_`), indicating that
+this is a Sass partial file. See the [Sass documentation][sass-partials] for more information about
+partial files.
+
+[sass-partials]: https://sass-lang.com/guide#topic-4
+
+```scss
+// carousel.scss
+
+.my-carousel {
+  display: flex;
+}
+
+.my-carousel-button {
+  border-radius: 50%;
+}
+```
 
 ```scss
-@mixin candy-carousel-theme($theme) {
-  // Extracts the color and typography configurations from the theme.
-  $color: mat-get-color-config($theme);
-  $typography: mat-get-typography-config($theme);
-
-  // Do not generate styles if configurations for individual theming
-  // systems have been explicitly set to `null`.
-  @if $color != null {
-    @include candy-carousel-color($color); 
+// _carousel-theme.scss
+
+@mixin color($theme) {
+  .my-carousel-button {
+    color: blue;
   }
-  @if $typography != null {
-    @include candy-carousel-typography($typography);
+}
+
+@mixin typography($theme) {
+  .my-carousel {
+    font-family: serif;
   }
 }
 ```
 
-See the [typography guide](https://material.angular.io/guide/typography) for more information on
-typographic customization.
+### Step 2: Use values from the theme
 
-## 2. Define all remaining styles in a normal component stylesheet
+Now that theme theme-based styles reside in mixins, we can extract the values we need from the
+theme passed into the mixins.
 
-Define all styles unaffected by the theme in a separate file referenced directly in the component's
-`styleUrl`.  This generally includes everything except for color and typography styles.
+```scss
+// _carousel-theme.scss
 
+@use 'sass:map';
+@use '~@angular/material' as mat;
 
-## 3. Include the theme mixin in your application
+@mixin color($theme) {
+  // Get the color config from the theme.
+  $color-config: mat.get-color-config($theme);
```

Comment:
No null checks shown here but then they're using in the theme mixin section, is that intentional?